pkg/directory/http.go

```diff
@@ -0,0 +1,117 @@
+package directory
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+)
+
+type handler struct {
+	router   *chi.Mux
+	provider Provider
+}
+
+// NewHandler creates a new Handler.
+func NewHandler(provider Provider) http.Handler {
+	h := &handler{provider: provider}
+	h.router = chi.NewMux()
+	h.router.Get("/*", func(w http.ResponseWriter, r *http.Request) {
+		err := h.serve(r.Context(), w)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+	})
+	return h
+}
+
+// ServeHTTP serves an HTTP request with directory users and groups.
+func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	h.router.ServeHTTP(w, r)
+}
+
+func (h *handler) serve(ctx context.Context, w http.ResponseWriter) error {
+	groups, users, err := h.provider.GetDirectory(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get directory data: %w", err)
+	}
+	w.Header().Set("Content-Type", "application/zip")
+	return encodeBundle(w, groups, users)
+}
+
+func decodeBundle(r io.Reader) (groups []Group, users []User, err error) {
+	bs, err := io.ReadAll(r)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to read zip file: %w", err)
+	}
+
+	zr, err := zip.NewReader(bytes.NewReader(bs), int64(len(bs)))
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to open zip file for reading: %w", err)
+	}
+
+	for _, file := range []struct {
+		name string
+		ptr  any
+	}{
+		{"pomerium.io/DirectoryGroup.json", &groups},
+		{"pomerium.io/DirectoryUser.json", &users},
```

Comment:
would be great to have that defined as public constants so they could be referred from the console